src/phoenix/experimental/evals/functions/binary.py

```diff
@@ -211,3 +243,60 @@ def _extract_rail(string: str, positive_rail: str, negative_rail: str) -> Option
     elif negative_pos != -1:
         return negative_rail
     return None
+
+
+def retrieval_concat_and_truncate(values, encoding_name="gpt-4", max_tokens=7100):
+    """This is designed to be used on a row of a Pandas Dataframe column.
+       value = ["chunk", "chunk", chunk"]
+       It concats the chunks in a list to a str, the column can be used for Q&A Eval.
+       df['retrieved_context'].apply(lambda x: retrieval_concat_and_truncate(x))
+       It makes sure the string can fit in a model / token <min> string size,
+       drops the tokens that don't fit.
```

Comment:
dropping the tokens feels scary - should consider a different approach like splitting and accumulating.